packages/core/saved-objects/core-saved-objects-base-server-internal/src/saved_objects_config.ts

```diff
@@ -42,6 +42,8 @@ export const savedObjectsMigrationConfig: ServiceConfigDescriptor<SavedObjectsMi
 const soSchema = schema.object({
   maxImportPayloadBytes: schema.byteSize({ defaultValue: 26_214_400 }),
   maxImportExportSize: schema.number({ defaultValue: 10_000 }),
+  // Conditionally set within config, dependening on the env. In prod: allow all access(default), in dev: false
+  allowHttpApiAccess: schema.boolean({ defaultValue: true }),
```

Comment:
I've gone with defaulting to the "old" behavior where all visible types (those not hidden) use the HTTP APIs.

Reply:
Is there a specific reason to depend on the environment and not distribution?

Instead of doing the overrides between the config and the env inside the code, why not just do it from the config defaults (like the example below). This makes the logic a lot more resilient and easier to understand. wdyt?

```
schema.conditional(
      schema.contextRef('dist'),
      true,
      schema.boolean({ defaultValue: false }),
      schema.boolean({ defaultValue: true })
)
```

Reply:
> Is there a specific reason to depend on the environment and not distribution?

I didn't think to use the distro rather, great idea thanks!
I'll change it and see how it improves the logic.

Reply:
@Bamieh I've made the change you requested in [`96bb95a` (#151512)](https://github.com/elastic/kibana/pull/151512/commits/96bb95af2a214c2b4cc5bf6582879ea57dc8a2be).
Could you take another look, please?